SimTracker/TrackAssociation/plugins/TrackTimeValueMapProducer.cc

```diff
@@ -0,0 +1,270 @@
+{
+  // setup resolution models
+  const std::vector<edm::ParameterSet>& resos = conf.getParameterSetVector("resolutionModels");
+  for( const auto& reso : resos ) {
+    const std::string& name = reso.getParameter<std::string>("modelName");
+    ResolutionModel* resomod = ResolutionModelFactory::get()->create(name,reso);
+    resolutions_.emplace_back( resomod );  
+
+    // times and time resolutions for general tracks
+    produces<edm::ValueMap<float> >(generalTracksName+name);
+    produces<edm::ValueMap<float> >(generalTracksName+name+resolution);
+    
+    //for gsf tracks
+    produces<edm::ValueMap<float> >(gsfTracksName+name);
+    produces<edm::ValueMap<float> >(gsfTracksName+name+resolution);
+  }
+  // get RNG engine
+  edm::Service<edm::RandomNumberGenerator> rng;
+  if (!rng.isAvailable()){
+    throw cms::Exception("Configuration")
+      << "TrackTimeValueMapProducer::TrackTimeValueMapProducer() - RandomNumberGeneratorService is not present in configuration file.\n"
+      << "Add the service in the configuration file or remove the modules that require it.";
+  }
+
+}
+
+void TrackTimeValueMapProducer::produce(edm::Event& evt, const edm::EventSetup& es) {
+  // get RNG engine
+  edm::Service<edm::RandomNumberGenerator> rng;  
+  auto rng_engine = &(rng->getEngine(evt.streamID()));
+
+  // get sim track associators
+  std::vector<edm::Handle<reco::TrackToTrackingParticleAssociator> > associators;
+  for( const auto& token : associators_ ) {
+    associators.emplace_back();
+    auto& back = associators.back();
+    evt.getByToken(token,back);
+  }
+
+  std::vector<float> generalTrackTimes, gsfTrackTimes;
+ 
+  //get track collections
+  edm::Handle<edm::View<reco::Track> > TrackCollectionH;
+  evt.getByToken(tracks_, TrackCollectionH);
+  const edm::View<reco::Track>& TrackCollection = *TrackCollectionH;
+
+  edm::Handle<edm::View<reco::Track> > GsfTrackCollectionH;
+  evt.getByToken(gsfTracks_, GsfTrackCollectionH);
+  const edm::View<reco::Track>& GsfTrackCollection = *GsfTrackCollectionH;
+
+  //get tracking particle collections
+  edm::Handle<TrackingParticleCollection>  TPCollectionH;
+  evt.getByToken(trackingParticles_, TPCollectionH);
+  //const TrackingParticleCollection&  TPCollection = *TPCollectionH;
+
+  /*
+  edm::Handle<TrackingVertexCollection>  TVCollectionH;
+  evt.getByToken(trackingVertices_, TVCollectionH);
+  const TrackingVertexCollection&  TVCollection= *TVCollectionH;
+  */
+
+  /*
+  std::cout << "Track size            = " << TrackCollection.size() << std::endl;
+  std::cout << "GsfTrack size         = " << GsfTrackCollection.size() << std::endl;
+  std::cout << "TrackingParticle size = " << TPCollection.size() << std::endl;
+  std::cout << "TrackingVertex size   = " << TVCollection.size() << std::endl;
+  */
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant